### PR TITLE
Feature enhancements

### PR DESCRIPTION
**Pull request by rajvidhimar**
## What does this PR do?

Adds feature enhancements to junos execution module.
## Changes

The execution module 'call_rpc' is changed to just 'rpc'. New arguments were also added in that module. Few changes are made in the 'rpc' module to add support for devices having an older junos image. 
## Tests written?

No

### CODE REVIEW

**Comment by cachedout** [comment]
@cro Can you please have a look at this?

**Comment by damon-atkins** [comment]
Re: Module rename.   RPC is a standard.  Does this module provide generic RPC support, or is it specific to a product.

**Comment by rajvidhimar** [comment]
@damon-atkins This entire module is specific to Juniper's product (Junos). As far as the function rpc is concerned, any valid rpc is supported. By valid I mean the ones which junos (Juniper's OS) will understand.

**Comment by cachedout** [comment]
I agree with @rajvidhimar. `junos.rpc` is a good name for this.

File: salt/modules/junos.py [hunk 1]
```diff
@@ -107,48 +104,84 @@ def call_rpc(cmd=None, *args, **kwargs):
...
+    except Exception as exception:
         ret['message'] = 'Execution failed due to "{0}"'.format(exception)
         ret['out'] = False
 
-    if 'dest' in op:
-        f = open(op['dest'], 'w')
-        f.write(ret['message'])
-        f.close()
+    if dest is not None:
+        with open(dest, 'w') as fp:
```
Comment (cachedout) [inline salt/modules/junos.py]: `salt.utils.fopen`, please

**Review by cachedout — COMMENTED** [review]
Minor change requested to use Salt's own file handling but the rest looks wonderful!

**Comment by rajvidhimar** [comment]
When will this pull request be merged? I would appreciate if its done fast. We have a demo upcoming.

**Pull request merged**